### PR TITLE
Normalize partedGetPartitionID output (bsc#1136744)

### PR DESCRIPTION
**Pull request by davidcassany**
This commit ensures the partedGetPartitionID output returns the
same partition ID codes with or without the presence of sgdisk
utility. It basically returns the sgdisk parsed value in lowercase.

Additionally this commit also makes sure that the partition stated
to be of `ef` type in `config.<mac>` is named `legacy`, to be consistent
with the partedGetPartitionID expectations.

### CODE REVIEW

**Comment by schaefi** [comment]
The coding looks good to me, but I think only a real live test can make sure nothing else breaks. I guess you are waiting for feedback from Vladimir.

Thanks much for you effort here

**Comment by davidcassany** [comment]
@schaefi yes that's why this PR is blocked with needinfo. Also I did not ask your review because of that, at first I want some real tests. I'll try reproduce it also, Vladimir said we will test it on Monday and gave me some hints to reproduce it. Vladimir has a different suggestions to fix it, but I am afraid to touch `partedGetParitionID` method return values for the default GPT case, I have no clue what else could break in OEM realm.

File: modules/KIWILinuxRC.sh [hunk 1]
```diff
@@ -10493,6 +10494,10 @@ function pxePartitionInputGeneric {
             partID=fd
             pname=lxroot
         fi
+        if [ $partID = "ef" ] && [ ! -z "$RAID" ];then
```
Comment (nadvornik) [inline modules/KIWILinuxRC.sh]: My test fails on this: [ ! -z "$RAID" ]

Comment (davidcassany) [reply modules/KIWILinuxRC.sh]: Damn sorry, just realized I did not push this change last Friday, this was supposed to be 

```bash
if [ $partID = "ef" ] && [ "$partMount" = "/boot/efi" ];then
```

**Comment by davidcassany** [comment]
@nadvornik with the current status the PXE image just worked fine with and without sgdisk inside the initrd. Could you test it again?

The solution I did is to trust partition names over the sgdisk output. I noted that the code does some assumptions about parted default values and it's GPT support that I believe are not valid anymore. For instance the code assumes that the default partition type is `83` however in my tests it is `07`, it also assumes that for GPT partitions parted can't set the partition type, thus it never sets the partition type partitions.

Because of that I believe the best approach to check the partition type is relaying on the partition name, which is completely set by KIWI (depending on its type) and tool independent. Trying to run parted in a consistent way with sgdisk and GPT partitions I believe would be error prone and lead to a relevant refactoring of the related code. Since this is a very sensitive portion of code, that can only be validated running the initrd, huge changes there should be avoided when possible.

**Comment by nadvornik** [comment]
LGTM

**Pull request merged**